parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -453,7 +467,14 @@ pub mod pallet {
 				config::FINALIZED_ROOT_INDEX,
 			)?;
 
+			let last_finalized_period =
+				Self::compute_current_sync_period(last_finalized_header.beacon_slot);
 			let current_period = Self::compute_current_sync_period(update.attested_header.slot);
+			ensure!(
+				(current_period == last_finalized_period ||
+					current_period == last_finalized_period + 1),
+				Error::<T>::InvalidFinalizedPeriodUpdate
+			);
```

Comment:
Awesome, thanks for these checks! They address https://linear.app/snowfork/issue/SNO-388. 😄